source/common/upstream/eds.cc

```diff
@@ -42,7 +42,7 @@ void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
     throw EnvoyException(fmt::format("Unexpected EDS resource length: {}", resources.size()));
   }
   const auto& cluster_load_assignment = resources[0];
-  if (cluster_load_assignment.cluster_name() != cluster_name_) {
+  if (!(cluster_load_assignment.cluster_name() == cluster_name_)) {
```

Comment:
I think this one can be fixed internally with some operator overloading. Thoughts?